### ISSUE TITLE
Fix the mentions table

### ISSUE THREAD

**Issue by emanuele45** [elkarte/Elkarte#1469]
The mentions table has few tricks that should be ironed out:
 1. id_msg should become id_target or something similar,
 2. the status column should be split in two columns (one for the status and one for the accessibility)
It was designed that way because it was introduced late in development and few problems were found after beta 1 was released, so in order to respect the "do not change db-schema during beta" idea I introduced those tricks.
Acceptable for the conditions that "forced" to have them, better fix it in 1.1.

I'm a little worried by breaking things by changing table schema, but I hope it is acceptable.

**Comment by emanuele45** [comment]
This was introduced in #1950
